### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by asifaminisonline**
### **Hello, In this Milestone I,**

**_Highlights_**

**1. _Create a Ruby on Rails application that allows the user to:_**

- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.
- Transactions list page

**2. _Sign up and login pages_**

- The user should be able to register in the app with full name, email, and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages

**3. _When the user logs in, they are presented with the categories page._**

- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

**4. _Transactions page_**

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**5. _"Add a new category" page_**

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**6. _"Add a new transaction" page_**

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

[Presentation video]()

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class ApplicationController < ActionController::Base
+  # before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
```
Comment (KayLemba) [inline app/controllers/application_controller.rb]: Please kindly remove unused code for it makes your code base look untidy. Please implement this change i all files where you have unused code.

File: app/models/group.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Group < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  # has_many :group_purchases, dependent: :destroy
+  # has_many :purchases, through: :group_purchases
```
Comment (KayLemba) [inline app/models/group.rb]: Please kindly remove unused code for it makes your code base look untidy

File: app/models/purchase.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Purchase < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  # has_many :group_purchases, dependent: :destroy
+  # has_many :groups, through: :group_purchases
```
Comment (KayLemba) [inline app/models/purchase.rb]: Please kindly remove unused code for it makes your code base look untidy

**Review by KayLemba — CHANGES_REQUESTED** [review]
Hi @akhror-valiev    👋
Good job so far working on the project requirements!
There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight 🎉

- [x] No linter errors. ✔️


- [x] Nice UI design
- [x] Tests are passing
- [x] No linter errors



Required Changes ♻️
Check the comments under the review in the inline comments.

### **VIDEO PRESENTATION**

- [x] You did an amazing job recording a video presentation  of your project but unfortunately it wasnt professional enough because you didnt mention and show a piece of code and feature you are most proud of creating in your project. So it would be nice if you recorded another video based on the guidelines below:


- [x] The student recorded a video presenting the features of the project.

- [x] The maximum length of the video is 5 minutes.

- [x] The video contains a clear and objective description about the project in professional English. The topic and purpose of the project are made clear.

- [x] The video contains a demo of the project, highlighting all the project features.

- [x] The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

NOTE: Please it is important to be professional at all times please pay attention to detail.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Ismailco — CHANGES_REQUESTED** [review]
#### Hi @asifaminisonline, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

giphy


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally


## Suggested changes
- [ ] As the previous reviewer mentioned kindly make sure to display a piece of code you're proud of making in the video presentation, the README file and folder structure is not considered a piece of code
_Check the comments under the review._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @asifaminisonline, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to record another video to show a piece of code you are proud of 👏

## Optional suggestions

- [ ] Kindly take note that this project PR was supposed to be created from a feature branch, requesting to merge into development.

gitflow


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**